lib/semantics/type.cc

```diff
@@ -269,34 +269,22 @@ std::ostream &operator<<(std::ostream &o, const DeclTypeSpec &x) {
   }
 }
 
-ProcInterface::ProcInterface(const ProcInterface &that)
-  : symbol_{that.symbol_} {
-  if (that.type_) {
-    set_type(*that.type_);
-  }
-}
-ProcInterface &ProcInterface::operator=(ProcInterface &&that) {
-  symbol_ = that.symbol_;
-  type_ = std::move(that.type_);
-  return *this;
-}
-
 void ProcInterface::set_symbol(const Symbol &symbol) {
   CHECK(!type_);
   symbol_ = &symbol;
 }
 void ProcInterface::set_type(const DeclTypeSpec &type) {
   CHECK(!symbol_);
-  type_ = std::make_unique<DeclTypeSpec>(type);
+  type_ = type;
 }
 
 std::ostream &operator<<(std::ostream &o, const ProcDecl &x) {
   return o << x.name_.ToString();
 }
 
 ProcComponentDef::ProcComponentDef(
-    const ProcDecl &decl, Attrs attrs, ProcInterface &&interface)
-  : decl_{decl}, attrs_{attrs}, interface_{std::move(interface)} {
+    const ProcDecl &decl, Attrs attrs, const ProcInterface &interface)
+  : decl_{decl}, attrs_{attrs}, interface_{interface} {
```

Comment:
The move semantics seems to be going away in a lot of places. Is that to fix bugs? Or copies of these various data are now needed?

Reply:
I did this because it was simpler in a few places to make copies and there was no real benefit to using move semantics for the Details classes.